### PR TITLE
Try to fix the "$s | $s | $s" bug

### PR DESCRIPTION
**Pull request by BevapDin**
This is supposed to fix http://smf.cataclysmdda.com/index.php?topic=10259.0 - I can't reproduce nor even test it, so is something for the Windows people out there.

The problem only appears on Windows (it works fine on my Linux, insert rant here). Other string formatting seems to work fine, so what is different for this specific place (the character info title)? Well, it calls `mvwprintw` (a native ncurses function, but on Windows there is no ncurses). On Windows, that function is implemented in cursesport.cpp, it calls `vsnprintf` for the formatting.

However, the main wrapper to the printf formating (`string_format`) does explicitly not call `vsnprintf` on Windows (see output.cpp), it calls `_vsprintf_p`. This has been introduced for the support of positional parameters in the format string (see 12541077ff81073f390db71d9a5436385b941baf).

tl;dr: `mvwprintw` on Windows used `vsnprintf` which does not support positional parameters in the format. Now it calls the commonly accept wrapper (works everywhere else).

Did I mention that I can't test it here? So please take this with exceptional care.

### CODE REVIEW

**Comment by qbradq** [comment]
Confirmed issue exists in current master head, both Tiles and Curses build. Build environment was msys2 with 64-bit tool chain.

**Comment by qbradq** [comment]
Build fails in msys2 with 64-bit toolchain, GGC version is 4.9.1.

```
C:/msys64/home/qbradq/Cataclysm-DDA/src/cursesport.cpp: In function 'int wprintw(WINDOW*, const char*, ...)':
C:/msys64/home/qbradq/Cataclysm-DDA/src/cursesport.cpp:467:44: error: invalid conversion from 'const char*' to 'char*' [-fpermissive]
     return printstring(win, printbuf.c_str());
                                            ^
C:/msys64/home/qbradq/Cataclysm-DDA/src/cursesport.cpp:377:12: note: initializing argument 2 of 'int printstring(WINDOW*, char*)'
 inline int printstring(WINDOW *win, char *fmt)
            ^
C:/msys64/home/qbradq/Cataclysm-DDA/src/cursesport.cpp: In function 'int mvwprintw(WINDOW*, int, int, const char*, ...)':
C:/msys64/home/qbradq/Cataclysm-DDA/src/cursesport.cpp:480:44: error: invalid conversion from 'const char*' to 'char*' [-fpermissive]
     return printstring(win, printbuf.c_str());
                                            ^
C:/msys64/home/qbradq/Cataclysm-DDA/src/cursesport.cpp:377:12: note: initializing argument 2 of 'int printstring(WINDOW*, char*)'
 inline int printstring(WINDOW *win, char *fmt)
            ^
C:/msys64/home/qbradq/Cataclysm-DDA/src/cursesport.cpp: In function 'int mvprintw(int, int, const char*, ...)':
C:/msys64/home/qbradq/Cataclysm-DDA/src/cursesport.cpp:493:48: error: invalid conversion from 'const char*' to 'char*' [-fpermissive]
     return printstring(mainwin, printbuf.c_str());
                                                ^
C:/msys64/home/qbradq/Cataclysm-DDA/src/cursesport.cpp:377:12: note: initializing argument 2 of 'int printstring(WINDOW*, char*)'
 inline int printstring(WINDOW *win, char *fmt)
            ^
C:/msys64/home/qbradq/Cataclysm-DDA/src/cursesport.cpp: In function 'int printw(const char*, ...)':
C:/msys64/home/qbradq/Cataclysm-DDA/src/cursesport.cpp:503:48: error: invalid conversion from 'const char*' to 'char*' [-fpermissive]
     return printstring(mainwin, printbuf.c_str());
                                                ^
C:/msys64/home/qbradq/Cataclysm-DDA/src/cursesport.cpp:377:12: note: initializing argument 2 of 'int printstring(WINDOW*, char*)'
 inline int printstring(WINDOW *win, char *fmt)
            ^
```

**Comment by BevapDin** [comment]
I fixed those errors, who expects a print function to take a non-const string, that is just crazy. 

And thanks for testing :+1:

**Comment by qbradq** [comment]
Confirmed this resolves the compile errors under mingw32-make, Also confirmed the PR resolves the issue in both Curses and Tiles builds.
